src/common/server.ts

```diff
@@ -203,3 +228,29 @@ export async function restartServer(
 
   return newLSClient;
 }
+
+async function getRuffVersion(settings: ISettings): Promise<[number, number, number]> {
+  let command;
+  if (settings.path[0]) {
+    command = `${settings.path[0]} ${RUFF_VERSION_ARGS.join(" ")}`;
+  } else {
+    command = `${
+      settings.interpreter[0]
```

Comment:
What's the reason we use `settings.interpreter` here directly over `getInterpreterFromSetting`?

Reply:
This is how `createNativeServer` and `createLegacyServer` get the interpreter path - the settings passed in are initialized from `getWorkspaceSettings`, which itself calls `getInterpreterFromSetting` to set `settings.interpreter`.